metadata/db.go

```diff
@@ -55,9 +55,16 @@ func WithPolicyIsolated(o *dbOptions) {
 	o.shared = false
 }
 
+func WithSharedNamespace(ns string) DBOpt {
+	return func(o *dbOptions) {
+		o.sharedNS = ns
+	}
+}
+
 // dbOptions configure db options.
 type dbOptions struct {
-	shared bool
+	shared   bool
+	sharedNS string
 }
```

Comment:
Is it possible to reject an invalid combination (e.g. shared=true, sharedNS="foobar") here? I think it would be hard since the order of WithXXX methods will be arbitrary.